File: year2/code/1dt905/a01-labs/lab2/src/com/sb224sc/game/Deck.java
package com.sb224sc.game;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Deck {
    private List<Card> deckOfCards;

    public Deck() {
        generateDeck();
    }

    private void generateDeck() {
        deckOfCards = new ArrayList<>();
        String[] suits = { "Spades", "Hearts", "Clubs", "Diamonds" };
        String[] ranks = { "2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace" };

        for (String suit : suits) {
            for (int i = 0; i < ranks.length; i++) {
                int value = i + 2;

                deckOfCards.add(new Card(suit, ranks[i], value));
            }
        }
    }

    public void shuffle() {
        Collections.shuffle(deckOfCards);
    }

    public void printTopAndBottomCard() {
        if (!deckOfCards.isEmpty()) {
            System.out.println("\nTop Card: " + deckOfCards.get(0));
            System.out.println("Bottom Card: " + deckOfCards.get(deckOfCards.size() - 1));
        } else {
            System.out.println("\nDeck is empty.");
        }
    }

    public Card drawCard() {
        if (!deckOfCards.isEmpty()) {
            return deckOfCards.remove(0);
        } else {
            System.out.println("\nDeck is empty.");
            return null;
        }
    }

    public List<Card> dealPokerHand(int amountCards) {
        List<Card> hand = new ArrayList<>();

        for (int i = 0; i < amountCards; i++) {
            Card card = drawCard();
            if (card != null) {
                hand.add(card);
            }
        }

        return hand;
    }

    public int sumOfTwoCards() {
        List<Card> hand = dealPokerHand(2);
        int sum = 0;

        for (Card card : hand) {
            sum += card.getValue();
        }

        return sum;
    }
}

File: soppa/1MA930 - Numeriska/1DT901/jw223rn_assign2/deck.py
from random import shuffle


def add_colour(x):
    cards = ["2", "3", "4", "5", "6", "7", "8", "9", "10",
             "Jack", "Queen", "King", "Ace"]
    cards = [n + x for n in cards]
    return cards


heart_cards = add_colour(" of Hearts")
club_cards = add_colour(" of Clubs")
spade_cards = add_colour(" of Spades")
diamond_cards = add_colour(" of Diamonds")

deck = heart_cards + club_cards + spade_cards + diamond_cards

shuffle(deck)

print("My hand:")
print(deck[0])
print(deck[1])
print(deck[2])
print(deck[3])
print(deck[4])
